OpenRA.Mods.Common/Traits/TunnelEntrance.cs

```diff
@@ -0,0 +1,65 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class TunnelEntranceInfo : ITraitInfo, IRulesetLoaded
+	{
+		[Desc("Cell radius to use as a staging area for unit entry and exit.")]
+		public readonly int Margin = 4;
+
+		[Desc("Offset to check for the corresponding map tunnel portal.")]
+		public readonly CVec Sensor = CVec.Zero;
+
+		[Desc("Unit vector facing into tunnel.")]
+		public readonly CVec Direction = new CVec(1, 0);
```

Comment:
It would need to be NorthEast, NorthWest, SouthEast, SouthWest, or more descriptively -X, +X, -Y, +Y (except that we can't use those symbols in the enums).  We would then need to index these into an array of CVec to recover the vector, making it overall more complicated than what we have now.